### PR TITLE
Include Jue library source as a part of Philips Hue binding (#2601)

### PR DESCRIPTION
**Pull request by ddudnik**
Source code of Jue library is included into Hue binding under `org.eclipse.smarthome.binding.hue.internal` package. Binary 'jue.jar' is removed and all code references are updated.
Issue #2601

### CODE REVIEW

**Comment by maggu2810** [comment]
If we want to keep the history we need to remember to not "squash and merge".

I think we could merge it as it is, all further changes on the lib needs to be done later if we get it into this repo.

**Comment by maggu2810** [comment]
I restarted the build because a thing test failed (don't think it is related to this PR).

**Comment by kaikreuzer** [comment]
> If we want to keep the history we need to remember to not "squash and merge".

I actually doubt that we want/can keep the history in place.
We so far have a CQ for a specific version of the library (and not for ALL versions up to that point), so whatever code might have existed earlier in history has NOT been IP approved.

We will also require the copyright holder to approve this contribution with adding him to the commit message. We therefore should imho have a single commit, which both the copyright holder and the Eclipse IP team can review and approve.

**Comment by maggu2810** [comment]
Sound reasable, okay.

**Comment by kaikreuzer** [comment]
@ddudnik Please let me know, once you have updated the PR. I will then try to reach out to the copyright holder for their approval.

**Comment by ddudnik** [comment]
@kaikreuzer I rebased on upstream/master and squashed all commits into one as requested.

**Comment by kaikreuzer** [comment]
Thanks @ddudnik I am taking over regarding further IP clarifications. Give me a week or two for that before we can continue.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/BridgeDiscovery.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+import org.eclipse.smarthome.binding.hue.internal.HttpClient.Result;
+import org.eclipse.smarthome.binding.hue.internal.exceptions.ApiException;
+
+import com.google.gson.Gson;
+
+/**
+ * Various ways of discovering bridges.
+ *
+ * @author Alexander Overvoorde - initial contribution (as standalone Jue library)
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/BridgeDiscovery.java]: Could you please replace this line (in all classes from the Jue library) by:
```
@author Q42, standalone Jue library (https://github.com/Q42/Jue)
```

**Comment by ddudnik** [comment]
@kaikreuzer done, '@author' tag changed.

**Comment by kaikreuzer** [comment]
Thanks @ddudnik. I have created CQ 12624, so now the ball is with the Eclipse IP team.

**Comment by kaikreuzer** [comment]
Good news, we have check-in approval!
Thanks @ddudnik for your work and patience - now the way is clear to implement cool new features for hue :-)

**Comment by ddudnik** [comment]
Glad to hear that!

**Pull request merged**